fs/inode/fs_files.c

```diff
@@ -721,17 +786,62 @@ int fs_getfilep(int fd, FAR struct file **filep)
 
   *filep = files_fget(list, fd);
 
-  /* if f_inode is NULL, fd was closed */
+  /* if *filep is NULL, fd was closed */
 
-  if ((*filep)->f_inode == NULL)
+  if (*filep == NULL)
     {
-      *filep = NULL;
       return -EBADF;
     }
 
   return OK;
 }
 
+/****************************************************************************
+ * Name: fs_putfilep
+ *
+ * Description:
+ *   Handles reference counts for files, less than or equal to 0 and close
+ *   the file
+ *
+ * Input Parameters:
+ *   filep  - The caller provided location in which to return the 'struct
+ *            file' instance.
+ ****************************************************************************/
+
+#ifdef CONFIG_FS_REFCOUNT
+int fs_putfilep(FAR struct file *filep)
+{
+  irqstate_t flags;
+  int ret = 0;
+  int refs;
+
+  DEBUGASSERT(filep);
+  flags = spin_lock_irqsave(NULL);
+
+  refs = --filep->f_refs;
```

Comment:
we cloud replace typs of f_refs to atomic in next patch

Reply:
diff --git a/include/nuttx/fs/fs.h b/include/nuttx/fs/fs.h
index 9b711fc1c5..b3fd149016 100644
--- a/include/nuttx/fs/fs.h
+++ b/include/nuttx/fs/fs.h
@@ -471,7 +471,7 @@ struct file
 {
   int               f_oflags;   /* Open mode flags */
 #ifdef CONFIG_FS_REFCOUNT
-  int               f_refs;     /* Reference count */
+  atomic_int        f_refs;     /* Reference count */
 #endif
   off_t             f_pos;      /* File position */
   FAR struct inode *f_inode;    /* Driver or file system interface */
Do you mean to revise it like this? Can you first approve this PR? I will submit a new one later, because this PR blocks many submissions already.thanks very mouch

Reply:
@anchao

Reply:
Yes, no need to support this feature in PR, just suggestion

Reply:
@Zhangshoukui please fix the comment here 
https://github.com/apache/nuttx/pull/13296?notification_referrer_id=NT_kwDOAAuS3bIxMjI2MjQ0ODg3Mzo3NTg0OTM&notifications_query=reason%3Amention#discussion_r1759669745